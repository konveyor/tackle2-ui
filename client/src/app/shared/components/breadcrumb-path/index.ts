export { BreadCrumbPath } from "./breadcrumb-path";
