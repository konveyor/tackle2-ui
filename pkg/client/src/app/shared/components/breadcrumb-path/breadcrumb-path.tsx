import React from "react";
import { Link } from "react-router-dom";
import { Breadcrumb, BreadcrumbItem, Button } from "@patternfly/react-core";

export interface BreadCrumbPathProps {
  breadcrumbs: { title: string; path: string | (() => void) }[];
}

export const BreadCrumbPath: React.FC<BreadCrumbPathProps> = ({
  breadcrumbs,
}) => {
  return (
    <Breadcrumb>
      {breadcrumbs.map((crumb, i, { length }) => {
        const isLast = i === length - 1;

        const link =
          typeof crumb.path === "string" ? (
            <Link className="pf-c-breadcrumb__link" to={crumb.path}>
              {crumb.title}
            </Link>
          ) : (
            <Button variant="link" isInline onClick={crumb.path}>
              {crumb.title}
            </Button>
          );

        return (
          <BreadcrumbItem key={i} isActive={isLast}>
            {isLast ? crumb.title : link}
          </BreadcrumbItem>
        );
      })}
    </Breadcrumb>
  );
};
