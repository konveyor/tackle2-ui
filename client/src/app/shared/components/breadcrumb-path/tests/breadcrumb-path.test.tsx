import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { mount, shallow } from "enzyme";
import { BreadCrumbPath } from "../breadcrumb-path";

describe("BreadCrumbPath", () => {
  it("Renders without crashing", () => {
    const wrapper = shallow(
      <BreadCrumbPath
        breadcrumbs={[
          {
            title: "first",
            path: "/first",
          },
          {
            title: "second",
            path: "/second",
          },
          {
            title: "thrid",
            path: "/thrid",
          },
        ]}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("Given a function path, should callback", () => {
    const secondBreadcrumbSpy = jest.fn();

    const wrapper = mount(
      <Router>
        <BreadCrumbPath
          breadcrumbs={[
            {
              title: "first",
              path: "/first",
            },
            {
              title: "second",
              path: secondBreadcrumbSpy,
            },
            {
              title: "thrid",
              path: "/thrid",
            },
          ]}
        />
      </Router>
    );

    wrapper.find("button").simulate("click");
    expect(secondBreadcrumbSpy).toHaveBeenCalledTimes(1);
  });
});
