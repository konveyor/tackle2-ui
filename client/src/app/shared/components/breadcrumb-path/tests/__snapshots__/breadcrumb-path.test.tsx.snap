// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadCrumbPath Renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        aria-label="Breadcrumb"
        class="pf-c-breadcrumb"
        data-ouia-component-id="OUIA-Generated-Breadcrumb-1"
        data-ouia-component-type="PF4/Breadcrumb"
        data-ouia-safe="true"
      >
        <ol
          class="pf-c-breadcrumb__list"
        >
          <li
            class="pf-c-breadcrumb__item"
          >
            <a
              class="pf-c-breadcrumb__link"
              href="/first"
            >
              first
            </a>
          </li>
          <li
            class="pf-c-breadcrumb__item"
          >
            <span
              class="pf-c-breadcrumb__item-divider"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 256 512"
                width="1em"
              >
                <path
                  d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                />
              </svg>
            </span>
            <a
              class="pf-c-breadcrumb__link"
              href="/second"
            >
              second
            </a>
          </li>
          <li
            class="pf-c-breadcrumb__item"
          >
            <span
              class="pf-c-breadcrumb__item-divider"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 256 512"
                width="1em"
              >
                <path
                  d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                />
              </svg>
            </span>
            thrid
          </li>
        </ol>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      aria-label="Breadcrumb"
      class="pf-c-breadcrumb"
      data-ouia-component-id="OUIA-Generated-Breadcrumb-1"
      data-ouia-component-type="PF4/Breadcrumb"
      data-ouia-safe="true"
    >
      <ol
        class="pf-c-breadcrumb__list"
      >
        <li
          class="pf-c-breadcrumb__item"
        >
          <a
            class="pf-c-breadcrumb__link"
            href="/first"
          >
            first
          </a>
        </li>
        <li
          class="pf-c-breadcrumb__item"
        >
          <span
            class="pf-c-breadcrumb__item-divider"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              height="1em"
              role="img"
              style="vertical-align: -0.125em;"
              viewBox="0 0 256 512"
              width="1em"
            >
              <path
                d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
              />
            </svg>
          </span>
          <a
            class="pf-c-breadcrumb__link"
            href="/second"
          >
            second
          </a>
        </li>
        <li
          class="pf-c-breadcrumb__item"
        >
          <span
            class="pf-c-breadcrumb__item-divider"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              height="1em"
              role="img"
              style="vertical-align: -0.125em;"
              viewBox="0 0 256 512"
              width="1em"
            >
              <path
                d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
              />
            </svg>
          </span>
          thrid
        </li>
      </ol>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
